RileyLinkBLEKit/RileyLinkBLEDevice.m

```diff
@@ -464,7 +464,7 @@ - (void) setCustomName:(nonnull NSString*)customName {
 }
 
 - (void) onIdle {
-    if (idleListeningEnabled && _peripheral.state == CBPeripheralStateConnected) {
+    if (!runningIdle && idleListeningEnabled && _peripheral.state == CBPeripheralStateConnected && dataCharacteristic != nil) {
```

Comment:
It was another getting out of sync issue, I believe; it may have even been a situation where it was repeatedly thinking it was getting interrupted, but the idle loop was interrupting itself.  I was doing a bunch of testing with the pump out of range to see what kinds of situations arose.  If it's an issue, I'm sure I'll be able to repeat it, and look at it more closely.  I was going to fast, trying to wrap things up last night.